src/dbnode/persist/fs/migration/migration.go

```diff
@@ -0,0 +1,101 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package migration
+
+import (
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/storage"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+// Migration interface is implemented by tasks that wish to perform a data migration
+// on a fileset. Typically involves updating files in a fileset that were created by
+// a previous version of the database client
+type Migration interface {
+
+	// Run is the set of steps to successfully complete a migration
+	Run() error
+}
+
+var _ Migration = &ToVersion1_1{}
+
+// ToVersion1_1 is an object responsible for migrating a fileset to version 1.1
+type ToVersion1_1 struct {
+	newMergerFn fs.NewMergerFn
+	infoFile    fs.ReadInfoFileResult
+	shard       uint32
+	namespace   namespace.Metadata
+	opts        storage.Options
+}
+
+func NewToVersion1_1(infoFile fs.ReadInfoFileResult, shard uint32, namespace namespace.Metadata, opts storage.Options) Migration {
+	return ToVersion1_1{
+		newMergerFn: fs.NewMerger,
+		infoFile:    infoFile,
+		shard:       shard,
+		namespace:   namespace,
+		opts:        opts,
+	}
+}
+
+// Run executes the steps to bring a fileset to Version 1.1
+func (v ToVersion1_1) Run() error {
+	reader, err := fs.NewReader(v.opts.BytesPool(), v.opts.CommitLogOptions().FilesystemOptions())
+	if err != nil {
+		return err
+	}
+
+	merger := v.newMergerFn(reader, v.opts.DatabaseBlockOptions().DatabaseBlockAllocSize(),
+		v.opts.SegmentReaderPool(), v.opts.MultiReaderIteratorPool(),
+		v.opts.IdentifierPool(), v.opts.EncoderPool(), v.opts.ContextPool(), v.namespace.Options())
+
+	volIndex := v.infoFile.Info.VolumeIndex
+	fsID := fs.FileSetFileIdentifier{
+		Namespace:   v.namespace.ID(),
+		Shard:       v.shard,
+		BlockStart:  xtime.FromNanoseconds(v.infoFile.Info.BlockStart),
+		VolumeIndex: volIndex,
+	}
+
+	nsCtx := namespace.NewContextFrom(v.namespace)
+
+	flushPersist, err := v.opts.PersistManager().StartFlushPersist()
+	if err != nil {
+		return err
+	}
+
+	// Intentionally use a noop merger here as we simply want to rewrite the same files with the current encoder which
+	// will generate index files with the entry level checksums.
+	close, err := merger.Merge(fsID, storage.NewNoopMergeWith(), volIndex+1, flushPersist, nsCtx,
+		&persist.NoOpColdFlushNamespace{})
```

Comment:
I'm passing in a `&persist.NoOpColdFlushNamespace{}` here because this appears to be the only thing implementing this interface? Is this a placeholder for something in the future or did I miss something?

Reply:
No this is fine 👍  It's a callback to act on a per-series event during a merge, here we don't need to do anything of that nature.

Reply:
Discussed offline.